### PR TITLE
vfs: add refresh flag

### LINKED ISSUE #6830: mount flag to run vfs/refresh recursive=true
#### The associated forum post URL from `https://forum.rclone.org`
n/a

#### What is your current rclone version (output from `rclone version`)?
1.6.1

#### What problem are you are trying to solve?
vfs/refresh w/o needing --rc

#### How do you think rclone should be changed to solve that?
add a flag to be used on mounts that would warm up the vfs cache.
I don't want to enable --rc because I don't need to except to do this at service startup.

A flag could do this.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

### PR DESCRIPTION
**Pull request by beyondmeat**
#### What is the purpose of this change?

Add vfs/refesh recursive=true flag so it can run w/o needing rc on mount.

#### Was the change discussed in an issue or in the forum before?

closes #6830 

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Review by ncw — APPROVED** [review]
Looks great - thank you :-)

I'll merge this now.

**Comment by ncw** [comment]
@beyondmeat can you commit the missing bit of this PR? I think you forgot to commit the `vfs/vfscommon/options.go` file.

**Comment by beyondmeat** [comment]
@ncw will get to this by end of weekend. I didn't finish that part yet.

**Comment by beyondmeat** [comment]
```
--vfs-refresh=true           Refreshes the directory listing recursively
```

I made the default value false. Should be ready for review @ncw

**Comment by darthShadow** [comment]
Can we add an option to do it async too? So perhaps instead of `true`/`false`, `off` (default)/`sync`/`async` ?

**Comment by beyondmeat** [comment]
Is there a reason why anyone would ever want the refresh to be blocking/sync?

**Comment by darthShadow** [comment]
The current behaviour is sync. I would prefer async always but not sure about others so may as well add compatibility with the rc way of doing it which is both sync and sync.

**Comment by beyondmeat** [comment]
I'm unsure if sync would ever be used. It would effectively block the mount until that process is done, which I doubt anyone would ever need.

The use-case for that implies the options recursive=true and _async=true

I would also need another flag to control if it's recursive or not if you just want it to match the --rc functionality, which I don't think it's neccessary b/c this only happens at mount and why wouldn't it be recursive?  The whole point of this flag is to pre-warm the dir cache at mount so you don't need --rc otherwise. Having it sync and non-recursive, I fail to think of a use-case where that would be needed, if it is, they might need --rc anyway.

I'm aware my current implementation is sync, I'm exploring how to make it async, any example code I could reference to help with that. I tried code searching for async but not sure how _async=true works in the code.

**Comment by darthShadow** [comment]
It runs the async jobs in a goroutine: https://github.com/rclone/rclone/blob/9a9ef040e3be057227b14d43c0be1549bf5b1d67/fs/rc/jobs/job.go#L281

I guess you could do the same here by calling the `readDirTree` function in a goroutine.

**Comment by ncw** [comment]
This has a build failure which needs fixing

```
Error: vfs/vfsflags/vfsflags.go:33:59: not enough arguments in call to flags.BoolVarP
	have (*pflag.FlagSet, *bool, string, string, string)
	want (*pflag.FlagSet, *bool, string, string, bool, string)
```

I'd be fine if this was always async so just run the call to read tree in a go routine to do that.

**Comment by beyondmeat** [comment]
@ncw is this good now?

File: vfs/vfs.go
Comment (darthShadow) [inline vfs/vfs.go:237]: This is still synchronous. Run it in a goroutine by adding `go` in front of the statement to make it async.

**Comment by darthShadow** [comment]
I have also approved the CI run so let's see if anything else needs to be changed.

**Comment by beyondmeat** [comment]
@darthShadow ok fixed that lint error, please run the checks again. Not sure why it's still complaining about an import.

**Comment by beyondmeat** [comment]
@ncw @darthShadow please check

**Comment by ncw** [comment]
@beyondmeat sorry this dropped off the radar! I rebased this - I'll wait for the CI to go green then merge :-)

**Comment by ncw** [comment]
Thank you :-)

**Pull request merged**
